main.tf

```diff
@@ -81,6 +80,42 @@ resource "azurerm_linux_function_app" "linux_function" {
   }
 }
 
+
+resource "azurerm_subnet" "main" {
+  count                = var.create_subnet && length(var.subnet_cidr) != 0 && length(var.subnet_name) == 0 ? 1 : 0
+  name                 = "SN-${var.function_app_name}"
+  virtual_network_name = var.vnet_name
+  address_prefixes     = var.subnet_cidr
+  resource_group_name  = var.vnet_rg_name
+
+  delegation {
+    name = "delegation"
+
+    service_delegation {
+      name    = "Microsoft.Web/serverFarms"
+      actions = ["Microsoft.Network/virtualNetworks/subnets/action"]
+    }
+  }
+}
+
+data "azurerm_subnet" "main" {
+  count                = length(var.subnet_name) != 0 ? 1 : 0
+  name                 = var.subnet_name
+  virtual_network_name = var.vnet_name
+  resource_group_name  = var.vnet_rg_name
+  depends_on = [
+    azurerm_subnet.main
+  ]
+}
+
+
+resource "azurerm_app_service_virtual_network_swift_connection" "linux" {
+  # count          = var.create_subnet && length(var.subnet_cidr) != 0 && length(var.subnet_name) == 0 && var.asp_os_type == "Linux" ? 1 : 0
+  count          = var.asp_os_type == "Linux" ? 1 : 0
```

Comment:
Looks like the connection created unconditionally, but not all FA will need to be network bound. Do you need to use the line above instead?